src/ray/raylet/worker_pool.cc

```diff
@@ -182,7 +182,9 @@ void WorkerPool::PopWorkerCallbackInternal(const PopWorkerCallback &callback,
                                            std::shared_ptr<WorkerInterface> worker,
                                            PopWorkerStatus status) {
   RAY_CHECK(callback);
-  auto used = callback(worker, status);
+  // This method shouldn't be used with RuntimeEnvCreationFailed.
```

Comment:
add more detail why this is the case

Reply:
```
    // This method shouldn't be invoked when runtime env creation has failed because
    // when runtime env is failed to be created, they are all
    // invoking the callback immediately.
```

I found many interface is pretty brittle in worker pool... Hopefully we can clean this up soon

Reply:
Alternatively, we can modify every function to receive error message as an argument